hooks/environment

```diff
@@ -12,6 +12,22 @@ check_command() {
   fi
 }
 
+# the buildkite agent redactor has been added from version 3.66.0 and on. So we need to check it.
+# https://github.com/buildkite/agent/releases/tag/v3.66.0
+check_buildkite_version() {
+  local version_output
+  version_output=$(buildkite-agent --version)
+  local version
+  version=$(echo "$version_output" | grep -oP '\d+\.\d+\.\d+')
+  local min_version="3.66.0"
+  if [ "$(printf '%s\n' "$min_version" "$version" | sort -V | head -n1)" = "$min_version" ]; then
+    return 0
+  else
+    echo "Buildkite agent version $version is less than $min_version. Redactor is not available."
```

Comment:
Warnings/errors should go to `stderr`:
```suggestion
    echo "Buildkite agent version $version is less than $min_version. Redactor is not available." >&2
```

Reply:
Same as below:
Maybe they should but this was not the case so far (https://github.com/elastic/vault-secrets-buildkite-plugin/blob/main/hooks/environment#L10) so I do not believe this is necessary to be done in the context of this PR. Maybe a new PR could be opened to change both of them.

Reply:
As below, we should be moving net-new code to better practices

Reply:
Same as below this is addressed in a clean way in https://github.com/elastic/vault-secrets-buildkite-plugin/pull/37/files which will be merged after this one.

Reply:
I still disagree, but in the interest of time, I'll let it pass

Reply:
I appreciate your feedback that's is why I created the new PR to address it.